rm-datasource/src/main/java/io/seata/rm/datasource/exec/mysql/MySQLInsertOnDuplicateUpdateExecutor.java

```diff
@@ -317,6 +329,9 @@ public String buildImageSQL(TableMeta tableMeta) {
                             }
                             break;
                         }
+                        if ("PRIMARY".equalsIgnoreCase(k)) {
+                            primaryKeysInBeforeImageSql.add(columnName);
```

Comment:
there may be an escape character issue here. Keywords require escape characters, such as' order '.